gui.py

```diff
@@ -0,0 +1,173 @@
+# gui.py
+# Uses PyQt to collect camera footage and display information from Microcontroller
+# Uses PyQt to collect keyboard input too
+
+import random
+import sys
+
+from PyQt5.QtCore import Qt, QObject, QThread, pyqtSignal, QTimer, QEvent
+import PyQt5.QtMultimedia as QTM
+import PyQt5.QtMultimediaWidgets as QTMW
+import PyQt5.QtWidgets as QT
+from PyQt5.QtGui import QKeyEvent
+
+from mcu import MCUInterface
+
+sys.path.insert(0, './mcu-lib')
+app = QT.QApplication(sys.argv)
+
+
+class MainWindow(QT.QWidget):
+    def __init__(self, mcuobject: MCUInterface, comms: Communications, exit_program: ExitProgram):
+        super().__init__()
+        self.mcu: MCUInterface = mcuobject
+        self.thruster1speed: int = 0
+        self.thruster2speed: int = 0
+        self.thruster3speed: int = 0
+        self.thruster4speed: int = 0
+        self.servospeed: int = 0
+        
+        self.comms: Communications = comms
+        self.exit_program: ExitProgram = exit_program
+        
+        # (key, message sent to comms)
+        self.keys_pressed = [
+            ("W", "lw"),
+            ("w", "w"),
+            ("A", "la"),
+            ("a", "a"),
+            ("S", "ls"),
+            ("s", "s"),
+            ("D", "ld"),
+            ("d", "d"),
+            ("E", "le"),
+            ("e", "e"),
+            ("Q", "lq"),
+            ("q", "q"),
+            (" ", "spacebar")
+        ]
+        
+        self.keys_released = [
+            ("w", "sw"),
+            ("a", "sa"),
+            ("s", "ss"),
+            ("d", "sd"),
+            ("e", "se"),
+            ("q", "sq")
+        ]
+    
+    def updatetext(self):
+        self.voltage_info.setText(str(self.mcu.latest_voltage))
```

Comment:
Create variables for `self.mcu.latest_voltage`, `self.mcu.latest_gyro`, etc. and use those.